src/app/clusters/icd-management-server/ICDManagementCluster.cpp

```diff
@@ -459,7 +452,8 @@ void MatterIcdManagementPluginServerInitCallback()
     AttributeAccessInterfaceRegistry::Instance().Register(&gAttribute);
 
     // Configure ICD Management
-    ICDManagementServer::Init(storage, symmetricKeystore, icdConfigurationData);
+    ICDManagementServer::GetInstance().Init(storage, symmetricKeystore, icdConfigurationData);
+    chip::Server::GetInstance().GetICDManager().RegisterObserver(&ICDManagementServer::GetInstance());
```

Comment:
We need to unregister the observer on shutdown, no?

Reply:
added, thanks